controller/internal/enforcer/applicationproxy/applicationproxy.go

```diff
@@ -261,7 +261,7 @@ func (p *AppProxy) registerAndRun(ctx context.Context, puID string, ltype common
 		c := httpproxy.NewHTTPProxy(p.collector, puID, caPool, appproxy, proxyMarkInt, p.secrets, p.registry)
 		return c, c.RunNetworkServer(ctx, listener, encrypted)
 	default:
-		c := tcp.NewTCPProxy(p.tokenaccessor, p.collector, puID, p.registry, p.cert, caPool)
+		c := tcp.NewTCPProxy(p.tokenaccessor, p.collector, puID, p.registry, p.cert, caPool, p.secrets)
 		return c, c.RunNetworkServer(ctx, listener, encrypted)
```

Comment:
Don't pass the secrets here. Do the calculation of the header you need and pass that. No reason to go over all the logic for every packet.

Reply:
fixed